914540466 modules cms content add unit tests be

KSergiy opened the pull request.

omairkhalidoverdose committed:
Unit tests for cms content module. https://www.wrike.com/open.htm?id=914540466

omairkhalidoverdose committed:
Unit tests for cms content module. https://www.wrike.com/open.htm?id=914540466

kovalevandrew committed:
added title and identifier to backup preview, change backup format to json https://www.wrike.com/open.htm?id=914538569

kovalevandrew committed:
added Apply button with confirm popup to recover page builder content and save block/page content https://www.wrike.com/open.htm?id=914533228

omairkhalidoverdose committed:
Added more tests for the module. https://www.wrike.com/open.htm?id=914540466

omairkhalidoverdose committed:
Added more tests for the module. https://www.wrike.com/open.htm?id=914540466

omairkhalidoverdose committed:
Code refactoring. Added docblock comments. Updated few test cases.https://www.wrike.com/open.htm?id=914540466

omairkhalidoverdose committed:
Code refactoring. Added docblock comments. Updated few test cases.https://www.wrike.com/open.htm?id=914540466

KSergiy commented on Test/Unit/Block/Adminhtml/Cms/CmsAbstractTest.php:
```diff
@@ -0,0 +1,243 @@
+     * @param int $id
+     * @param string $bcType
+     */
+    public function testGetCmsObject(int $id, string $bcType)
```
@omairkhalidoverdose this test is not useful, the idea of function is returning a CMS object according to $bcType so in the test, you must pass the variable into the function ( in this case object ) and get the expected result.

From my side better to do it in 2 variants:
- create 2 separate tests for history blocks and there test getCmsObject() function
- or via Reflection set different $bcType value for each test iteration and then test getCmsObject() function

KSergiy commented on Test/Unit/Block/Adminhtml/Cms/CmsAbstractTest.php:
```diff
@@ -0,0 +1,243 @@
+     * Test
+     * @return void
+     */
+    public function testGetBackupUrl()
```
@omairkhalidoverdose please move out all params into data provider and made at least 2 possible cases with different $bcType

And for the future: writing tests for such functions are useful because the function makes simple functionality.

The function must be refactored by:
- adding check is passed param is the array
- is all passed data present and correct
- add some exceptions 
and only then will be some reason to write test for it.

KSergiy commented on Test/Unit/Model/StoreManagementTest.php:
```diff
@@ -0,0 +1,99 @@
+    /**
+     * @return void
+     */
+    public function testGetStoreIdsByCodes()
```
@omairkhalidoverdose move params into data provider and make at least 2 or 3 cases, with admin, without admin, ...

KSergiy commented on Test/Unit/Model/StoreManagementTest.php:
```diff
@@ -0,0 +1,99 @@
+    public function testFilterStoresByStoreCodes()
+    {
+        $storeCodes = ['admin', 'nz'];
+        $this->storeRepositoryMock->expects($this->atLeastOnce())
```
@omairkhalidoverdose there better to change to ->once() it will be more correct

KSergiy commented on Test/Unit/Model/StoreManagementTest.php:
```diff
@@ -0,0 +1,99 @@
+    /**
+     * @return void
+     */
+    public function testFilterStoresByStoreCodes()
```
@omairkhalidoverdose same, params move to data provider and make a few cases

KSergiy commented on Test/Unit/Controller/Adminhtml/Block/MassExportTest.php:
```diff
@@ -0,0 +1,183 @@
+    /**
+     * Initialize test
+     */
+    public function setUp(): void
```
@omairkhalidoverdose this test failing on init CollectionFactory mock

omairkhalidoverdose replied:
@KSergiy I can not replicate this issue on my local. Not sure if it's due to php version. I tested on 7.4 and 8.1.

orthener committed:
Merge branch '914533228_apply_button' into release

omairkhalidoverdose committed:
Improved unit tests by adding more test cases and improving checking logic. https://www.wrike.com/open.htm?id=914540466

omairkhalidoverdose committed:
Improved unit tests by adding more test cases and improving checking logic. https://www.wrike.com/open.htm?id=914540466

orthener committed:
Code formating. Set version. https://www.wrike.com/open.htm?id=914538569

- https://www.wrike.com/open.htm?id=914538569
- https://www.wrike.com/open.htm?id=914538569

orthener committed:
Remived unused. Update documantation. https://www.wrike.com/open.htm?id=914533228

orthener committed:
Merge branch 'release'

omairkhalidoverdose committed:
Fix for php8.1 compatibility issue in code.https://www.wrike.com/open.htm?id=914540466

omairkhalidoverdose committed:
Fixed issues after rebasing from master. Updated CHANGELOG.md . Fixed php8.1 compatibility issue. https://www.wrike.com/open.htm?id=914540466

omairkhalidoverdose committed:
Merge branch '914540466_modules_cms_content_add_unit_tests_be' of github.com:Overdose-Digital/magento2-cms-management into 914540466_modules_cms_content_add_unit_tests_be

omairkhalidoverdose committed:
Updated module version to 2.0.2 in composer.json https://www.wrike.com/open.htm?id=914540466
